homeassistant/components/devialet/coordinator.py

```diff
@@ -27,7 +30,16 @@ def __init__(self, hass: HomeAssistant, client: DevialetApi) -> None:
             update_interval=SCAN_INTERVAL,
         )
         self.client = client
+        self.entry = entry
+        self.hass = hass
 
     async def _async_update_data(self) -> None:
         """Fetch data from API endpoint."""
         await self.client.async_update()
+
+        if await self.client.async_search_allowed():
+            self.entry.async_create_background_task(
+                hass=self.hass,
+                target=self.client.async_discover_upnp_device(),
+                name=f"{DOMAIN}_UPnP_Discovery",
+            )
```

Comment:
Why do we do this?

Reply:
At every restart of the speaker, the UPnP will change. This method finds the UPnP port and initiates the playback capability.